### PR TITLE
[bphh-1801] Collaboration modeling

### PR DESCRIPTION
**Pull request by taw27**
## Description
[bphh-1801] Add migrations to add collaborator and permit_collaboration models

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-1801

## Steps to QA
N/A

### CODE REVIEW

File: app/models/permit_collaboration.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class PermitCollaboration < ApplicationRecord
+  belongs_to :collaborator
+  belongs_to :permit_application
+
+  enum collaboration_type: { submission: 0, review: 1 }
+  enum collaborator_type: { delegatee: 0, assignee: 1 }, _default: 0
```
Comment (bkmaibach) [inline app/models/permit_collaboration.rb:6]: delegate is probably a better word than delagatee

Comment (bkmaibach) [reply app/models/permit_collaboration.rb:6]: In regular parlance, a delegate refers to a person chosen or elected to represent a group, organization, or constituency, typically at a conference, convention, or other formal meeting. Delegates are given the authority to act, speak, or vote on behalf of the people they represent. The term can also refer to someone assigned specific responsibilities or tasks by someone in a position of authority.

File: app/models/collaborator.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Collaborator < ApplicationRecord
+  belongs_to :user
+  belongs_to :collaboratorable, polymorphic: true
```
Comment (bkmaibach) [inline app/models/collaborator.rb:3]: collaborator...able

Comment (bkmaibach) [reply app/models/collaborator.rb:3]: collaborable?

Comment (bkmaibach) [reply app/models/collaborator.rb:3]: Neither are words, so it's up to you

File: app/models/permit_collaboration.rb (hunk 1 above)
Comment (taw27) [reply app/models/permit_collaboration.rb:6]: @bkmaibach seems like a synonymn https://www.merriam-webster.com/thesaurus/delegatee

**Pull request merged**